File: Typscript/Step-1 Basic/Maths/printAllDivisor.ts
// Time Complexity: O(n)
function printAllDivisor(n:number) {
    for(let i=1; i<n; i++) {
        if(n%i==0){
            console.log(i);
        }
    }
}

// Time Complexity: O(sqrt(n))
function printAllDivisorBetter(n:number) {
    for(let i=1; i*i<=n; i++) {
        if(n%i==0) {
            console.log(i);
            if(n/i != i) {
                console.log(n/i);
            }
        }
    }
}

// printAllDivisor(36)
printAllDivisorBetter(36)

File: Go/Step1/Maths/printAllDivisor.go
package main

import "fmt"

// Time Complexity: O(N)
func printAllDivisor(num int) {
	for i := 1; i <= num; i++ {
		if num%i == 0 {
			fmt.Println(i)

		}
	}
}

// Time Complexity: O(sqrt(N))
func printAllDivisorBetter(num int) {
	for i := 1; i*i <= num; i++ {
		if num%i == 0 {
			fmt.Println(i)
			if num/i != i {
				fmt.Println(num / i)
			}
		}
	}
}
